analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -80,10 +80,15 @@ public static SyntaxNode GetSelfOrTopParenthesizedExpression(this SyntaxNode nod
         public static SyntaxNode GetFirstNonParenthesizedParent(this SyntaxNode node) =>
             node.GetSelfOrTopParenthesizedExpression().Parent;
 
-        public static IEnumerable<AttributeSyntax> GetAttributes(this SyntaxList<AttributeListSyntax> attributeLists,
-            KnownType attributeKnownType, SemanticModel semanticModel) =>
-            attributeLists.SelectMany(list => list.Attributes)
-                .Where(a => semanticModel.GetTypeInfo(a).Type.Is(attributeKnownType));
+        public static IEnumerable<AttributeSyntax> GetAttributes(this SyntaxList<AttributeListSyntax> attributeLists, KnownType attributeKnownType, SemanticModel semanticModel)
+        {
+            const string Suffix = nameof(Attribute);
+            var withoutSuffix = attributeKnownType.ShortName.EndsWith(Suffix)
+                ? attributeKnownType.ShortName.Remove(attributeKnownType.ShortName.Length - Suffix.Length)
+                : attributeKnownType.ShortName;
+            return attributeLists.SelectMany(x => x.Attributes)
+                .Where(x => x.Name.ToStringContains(withoutSuffix) && semanticModel.GetTypeInfo(x).Type.Is(attributeKnownType));
```

Comment:
I think you should use IsKnownType extension method to avoid duplicating its behavior. 

Please check also the `AttributeSyntaxExtensions.IsKnownType`.

Reply:
Awesome! I had a feeling that we have something for it, but somehow didn't find it.